OneSignalSDK/onesignal/core/src/main/java/com/onesignal/user/internal/operations/impl/executors/SubscriptionOperationExecutor.kt

```diff
@@ -39,6 +41,7 @@ internal class SubscriptionOperationExecutor(
     private val _configModelStore: ConfigModelStore,
     private val _buildUserService: IRebuildUserService,
     private val _newRecordState: NewRecordsState,
+    private val _iamFetchConsistencyManager: IConsistencyManager<IamFetchOffsetKey>,
```

Comment:
We're registering the consistency manager in the [user module](https://github.com/OneSignal/OneSignal-Android-SDK/pull/2168/files#diff-c2aa9f3964fe4a5f55d6cccc6769bcd58c5a6dd2a57b829141805b57eeb1cb2fR41), which is in core.

Since the consistency manager base class and related code is already in core, I figured might as well put all consistency related code together (e.g. enums & conditions).

The IAM manager reads the offset while the executors write the offset. Do you have any suggestions regarding how to not have both modules share the dependency?

Reply:
hmm, ya I can see our reflection implementation not liking the generic.

Reply:
The main thing I is that would like to not add is references to other modules in core. I think `IamFetchOffsetKey` might just be to specific of a name, as it could be used in the future for other things. Maybe rename this to `FetchOffsetKey` or `UserFetchOffsetKey`?